rcl/src/rcl/graph.c

```diff
@@ -336,7 +336,19 @@ rcl_get_node_names(
     rcl_node_get_rmw_handle(node),
     node_names,
     node_namespaces);
-  return rcl_convert_rmw_ret_to_rcl_ret(rmw_ret);
+
+  if (RMW_RET_OK != rmw_ret) {
+    return rcl_convert_rmw_ret_to_rcl_ret(rmw_ret);
+  }
+
+  // Check that none of the node names are NULL or empty
+  for (size_t i = 0u; i < node_names->size; ++i) {
```

Comment:
what about the node_namespaces? no need to check?

Reply:
Good point. I think empty namespaces are okay, but we can add a guard against NULL values.
See a4b89a1